### DISCUSSION TITLE
Change for multi-agent support (Automatically add dockge nodes)

### DISCUSSION THREAD

**Discussion by Horizontalized** [louislam/dockge#421] [Feature Request]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

API

### 🔖 Feature description

I have an idea on how this should work. I will reefer to the different multi dockge containers as following; Node (worker, where all the other containers are.) & Server (the main dockage UI, where we manage the "VMs" from.)
I understand this is a new addon to the dockge, would love this to work the way my theory is thought out.

Objective; Add dockge workers to dockge server. When they are added they should be displayed with hostname instead of IP but with the hostname clickable forwarding us with the local IP address to the node. 

Todays reality; We add dockge nodes from dockge server, showing the local ip address of the hostname. (preventing any automation of adding nodes automatically)

Reason for change; Templates, easier to run templates and spin up more VMs and containers running dockage nodes. (Automatic process)

A big plus would be having environmental variables pointing dockge nodes to the dockge server directly instead of having to do the first sign up and manual adding ip address and credentials. (Helpful when spinning up new VM's with ansible)

### ✔️ Solution

1. Adding process should be done from the nodes end. Preferably with an option of docker environment variables for automation (example ansible) instead of running templates that is already signed in and has the right creds and ip.
2. Server dockge should display the newly added node with its hostname instead of IP with a clickable hyper link of nodes IP address.

**Comment by rkroelin** [comment]
As I commented on a similar request, I'd also like this functionality.  It would be super handy and something like this is what drove me away from portainer (prohibitively expensive to access a feature like this for small users)

**Comment by ahmd-soliman** [comment]
+1. I set up Docker hosts automatically, and adding each one by hand in the primary's UI is the step I can't script. Letting a node register itself through an environment variable (primary URL plus a token) would fit Ansible or Terraform setups.
